src/exercises/exerciseTypes/orderWords/OrderWords.js

```diff
@@ -22,57 +21,424 @@ export default function OrderWords({
 
   const [initialTime, setInitialTime] = useState(new Date());
-  //const [buttonOptions, setButtonOptions] = useState(null);
   const [resetCounter, setResetCounter] = useState(0);
   const [hintCounter, setHintCounter] = useState(0);
   const [totalErrorCounter, setTotalErrorCounter] = useState(0);
   const [wordsReferenceStatus, setWordsReferenceStatus] = useState([]);
-  const [messageToAPI] = useState("");
+  const [messageToAPI, setMessageToApi] = useState("");
   const [exerciseContext, setExerciseContext] = useState("");
   const [clueText, setClueText] = useState([]);
-  const [translatedText, setTranslatedText] = useState("");
+  const [exerciseText, setExerciseText] = useState("");
   const [userSolutionWordArray, setUserSolutionWordArray] = useState([]);
   const [confuseWords, setConfuseWords] = useState();
   const [wordSelected, setWordSelected] = useState();
   const [posSelected, setPosSelected] = useState("");
-  const [wordSwapId, setWordSwapId] = useState(NO_WORD_SELECTED_ID);
-  const [wordSwapStatus, setWordSwapStatus] = useState("");
   const [solutionWords, setSolutionWords] = useState([]);
   const [isCluesRowVisible, setIsCluesRowVisible] = useState(false);
   const [isResetConfirmVisible, setIsResetConfirmVisible] = useState(false);
   const [isHandlingLongSentences, setIsHandlingLongSentences] = useState(true);
   const [isSentenceTooLong, setIsSentenceTooLong] = useState(false);
-  const [textBeforeTranslatedText, setTextBeforeTranslatedText] = useState("");
-  const [textAfterTranslatedText, setTextAfterTranslatedText] = useState("");
-  
-  console.log("Running ORDER WORDS EXERCISE")
+  const [textBeforeExerciseText, setTextBeforeExerciseText] = useState("");
+  const [textAfterExerciseText, setTextAfterExerciseText] = useState("");
+  const [movingObject, setMovingObject] = useState();
+
+  const solutionDragIndex = useRef();
+  const solutionDragOverIndex = useRef();
+  const isDragPlacementLeft = useRef();
+  const wordSoupDrag = useRef();
+  const moveOverElement = useRef();
+  const moveLastPosition = useRef();
+  const moveElement = useRef();
+  const moveElementInitialPosition = useRef();
+  const scrollY = useRef();
+
+  const handleTouchScroll = () => {
+    var prevElement = document.getElementById("orderExercise");
+    var currentElement = prevElement;
+    while(currentElement != null){
+      prevElement = currentElement;
+      if (prevElement.scrollTop != 0){
+        break;
+      }
+      currentElement = currentElement.parentNode;
+    }
+    if (prevElement.scrollTop != null) scrollY.current = prevElement.scrollTop;
+    else scrollY.current = 0;
+  }
+
+  function _getCurrentExerciseTime() {
+    let pressTime = new Date();
+    if (IS_DEBUG) {
+      console.log(pressTime - initialTime);
+      console.log("^^^^ time elapsed");
+    }
+    return pressTime - initialTime;
+  }
+
+  // Touch / Mouse Drag n' Drop
+  // Helper Functions
+
+  function _setMoveInitialPosition(x, y) {
```

Comment:
I think that's a good idea. Initially, I had placed it outside because I was wondering if I would also do this with the drag n' drop for the Desktop events, but I decided against it. Alternatively, I could also just set the variables and remove the function.